ext/mysqlnd/mysqlnd_wireprotocol.c

```diff
@@ -267,6 +267,24 @@ mysqlnd_read_header(MYSQLND_PFC * pfc, MYSQLND_VIO * vio, MYSQLND_PACKET_HEADER
 		pfc->data->packet_no++;
 		DBG_RETURN(PASS);
 	}
+	// @see https://dev.mysql.com/worklog/task/?id=12999
+	if (header->size > 0) {
+		DBG_ERR_FMT("Logical link: try reap pending data. Packet size=%zu", header->size);
+		zend_uchar *buf = emalloc(header->size);
+		if ((PASS == pfc->data->m.receive(pfc, vio, buf, header->size, conn_stats, error_info)) && buf[0] == ERROR_MARKER) {
+			php_mysqlnd_read_error_from_line(buf + 1, header->size - 1,
+			                                 error_info->error, sizeof(error_info->error),
+			                                 &error_info->error_no, error_info->sqlstate
+			);
+			if (error_info->error_no == ER_CLIENT_INTERACTION_TIMEOUT) {
+				efree(buf);
+				DBG_RETURN(FAIL);
+			} else {
+				error_info->error_no = 0;
```

Comment:
Why are you setting it to 0? I don't think I understand this bit of logic. Could you explain it to me please?

Reply:
I'm considering handling only one type of error. If it's not this type of error, I'll set the code to 0. However, this change might not be suitable.